### PR TITLE
Added JSON support to read baselines from pipeline

### LINKED ISSUE #845: Add JSON support for reading baselines from pipeline
Extending from #833, we can add JSON support for reading baselines from JSON configurations. Currently baselines can be exported to JSON, but not read from JSON.

We can support `.json` and `.jsonc` extensions.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #845

Also added `.yml` for source discovery.

Alot of this code should help prepare introducing support for Selectors, Module configs and Rules, but kept those changes out for now so PR doesn't get too big. 

Thought it would be easier to getting this in first with baselines 😄 .

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Code changes**
  - [x] Have unit tests created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Microsoft/PSRule/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule/Host/HostHelper.cs [hunk 1]
```diff
@@ -97,7 +103,12 @@ internal static void ImportResource(Source[] source, RunspaceContext context)
             if (source == null || source.Length == 0)
                 return;
 
-            Import(ReadYamlObjects(source, context), context);
+            var results = new List<ILanguageBlock>();
+
+            results.AddRange(ReadYamlObjects(source, context));
+            results.AddRange(ReadJsonObjects(source, context));
```
Comment (ArmaanMcleod) [inline src/PSRule/Host/HostHelper.cs:109]: Wasn't sure if this is what we should do, but basically combining YAML and JSON result blocks. Did the same for baselines.

Comment (BernieWhite) [reply src/PSRule/Host/HostHelper.cs:109]: I think that is fine.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@ArmaanMcleod Nice work. Happy with the changes except for one point.

Currently YAML schema support is provided by PSRule-vscode, this works great using the current schema. If we were to do this as suggested then the schema wouldn't work with JSON.

The two options I can see working would be:

1. Update the schema to support an array of resources, and test .Rule.yaml can also accept an array or resources instead of separate documents.
2. Add an additional schema that can be used for JSON/ JSONC that references the original language schema.

**Comment by BernieWhite** [comment]
Working example here: https://github.com/BernieWhite/PSRule-vscode/tree/bewhite/json-resources-support

**Comment by ArmaanMcleod** [comment]
@BernieWhite To me in this case creating a new schema for JSON/JSONC seems like an easier solution, like what you've done above. I personally thinking the `---` document structure for YAML is great already.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod I'm fine with that. I've added the schema. So should be good to merge.

**Pull request merged**